app/controllers/concerns/application_shared.rb

```diff
@@ -15,4 +15,35 @@ def set_timezone
     # Reset timezone for the next thread
     Time.zone = default_timezone
   end
-end
\ No newline at end of file
+
+  private
+
+  # Get the allowed parameters from the Apipie documentation in the API controllers.
+  def safe_params
+    apipie_params = Apipie.app.get_method_description("v2##{controller_name}", action_name).to_json[:params]
```

Comment:
Also, how will this work with namespaced controllers?  (Both isolated and non-isolated engines.)